File: NOVEMBER 24/09 November/Minimum sum gfg.cpp
class Solution {
  public:
    string minSum(vector<int> &arr) {
        // code here
        size_t len = arr.size();
        sort(arr.begin(), arr.end());
        int index = arr.size()-1;
        int carry = 0;
        string sum = "";
        int digit = 0;
        for(; index>0; index-=2)
        {
            digit = arr[index]+arr[index-1]+carry;
            carry = digit/10;
            digit = digit%10;
            sum.push_back('0'+digit);
        }
        if(index==0)
        {
            digit = arr[index] + carry;
            sum.push_back('0'+digit%10);
            carry = digit/10;
        }
        if(carry)
            sum.push_back('1');
        // removing trailing zeroes
        while(sum.size() && sum.back()=='0')
            sum.pop_back();
        // zero sum
        if(sum.size()==0)
            return "0";
            
        reverse(sum.begin(), sum.end());
        return sum;
    }
};

File: JUNE 2025/23 June/Minimum sum gfg.java
class Solution {
    String minSum(int[] arr) {
        // code here
        Arrays.sort(arr);
        StringBuilder sb = new StringBuilder();
        int carry = 0, i;
        for(i=arr.length-1; i>0; i-=2)
        {
            carry += (arr[i]+arr[i-1]);
            if(carry!=0)
                sb.append((char)('0'+carry%10));
            carry /= 10;
        }
        if(i>=0 && (carry+arr[0])>0)
        {
            carry += arr[0];
            sb.append((char)('0'+carry%10));
            carry /= 10;
        }
        if(carry>0)
        {
            sb.append((char)('0'+carry));
        }
        return sb.reverse().toString();
    }
}
